### PR TITLE
[WIP - do not merge] Last monster PR, I promise! Changes to symptoms, add allergies, etc.

### PR DESCRIPTION
**Pull request by teganmaharaj**
# Description

 - add allergies
 - made sure Tristan's issues (asymptomatic and getting really sick) were fixed
 - renamed "reported" symptoms to obs_symptoms to match the rest
 - symptoms and plateau depend on initial viral load
 - possibly other bugfixes and things I've forgotten. I'll do better at documenting these with commits in future.
 - switch back looping over humans in "at" for two reasons:
  1) I believe removing humans after checking them could create systematically lower risk of infection because of timing (it would prematurely remove the human before the end of their supposed duration at that place, meaning they would not be counted as a potential source of infection for humans who arrive just after)
  2) couldn't think of a clean way to do additive viral load when computing encounters symmetrically (possibly could be computed once for the whole location? But I think the timing doesn't work out well...)

Touches simulator, utils, and config.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected) (unclear what "expected" means at this point IMO)
- [x]  This change requires a documentation update (don't they all)

# How Has This Been Tested?

This breaks one of the tests/computation of metrics with a numpy error about not being able to compute things for NoneType.
I think it needs to check if things are None and if so, treat them as zero. Or maybe we should make some things that are None zero instead? 
I wasn't sure and wanted to discuss with someone who wrote the tests/metrics.

I'd also like to go over the infection logic with Prateek and maybe others.

# Checklist:

- [maybe. not sure what they are.] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [NO ] I have added tests that prove my fix is effective or that my feature works
- [NO] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: simulator.py [hunk 1]
```diff
@@ -72,7 +72,12 @@ def __init__(self, env, name, age, rng, infection_timestamp, household, workplac
         else:
             self.carefulness = (round(self.rng.normal(25, 10)) + self.age/2) / 100
 
+        # allergies
+        self.has_allergies = self.rng.rand() < P_ALLERGIES
+        len_allergies = self.rng.normal(1/self.carefulness, 1)
+        self.len_allergies = 1 if len_allergies < 1 else round(len_allergies)
```
Comment (pg2455) [inline simulator.py:78]: Do we need allergies?? Is it relevant for covid?

File: simulator.py [hunk 2]
```diff
@@ -618,71 +674,92 @@ def at(self, location, city, duration):
...
                     p_infection = h.infectiousness * ratio * (1-mask_efficacy) * proximity_factor # &prob_infectious
-                    x_human = self.rng.random() < p_infection * CONTAGION_KNOB
+                    initial_viral_load += p_infection
+                    x_human = self.rng.random() < initial_viral_load * CONTAGION_KNOB
 
+                    # Infection
                     if x_human and self.is_susceptible:
                         self.infection_timestamp = self.env.timestamp
+                        if initial_viral_load > VIRAL_LOAD_NORMALIZATION:
+                            self.initial_viral_load = 1
```
Comment (pg2455) [inline simulator.py:728]: How is incremental viral load different than the earlier version in which we had the precomputed viral load? Why is this necessary?

File: simulator.py [hunk 3]
```diff
@@ -618,71 +674,92 @@ def at(self, location, city, duration):
...
+                #     if x_human and h.is_susceptible:
+                #         h.infection_timestamp = self.env.timestamp
+                #         self.n_infectious_contacts+=1
+                #         Event.log_exposed(h, self, self.env.timestamp)
+                #         city.tracker.track_infection('human', from_human=self, to_human=h, location=location, timestamp=self.env.timestamp)
+                #         # print(f"{self.name} infected {h.name} at {location}")
+                #         infectee = h.name
+
+                # Encountered human is infectious
+                if h.is_infectious:
```
Comment (pg2455) [inline simulator.py:718]: I think this will miss a case when `self` is infectious. This will not let self infect other humans already at the location.

File: simulator.py (hunk 1 above)
Comment (teganmaharaj) [reply simulator.py:78]: It's like cold/flu; another source of symptoms. I thought it might be sort of important/useful to have one of the sources be totally random / not communicable.

File: simulator.py (hunk 2 above)
Comment (teganmaharaj) [reply simulator.py:728]: This is *INITIAL* viral load, i.e. the amount of virus that self is exposed to. The magnitude of this payload (also called "viral dose") influences the course of the disease, specifically if you're exposed to a larger amount of virus at infection, you are more likely to have severe symptoms and longer recovery (because your immune system is more immediately overwhelmed, and you have a harder time fighting off the disease overall.

The viral load curve is still precomputed based on this initial viral load, it's just precomputed at the time of infection now, instead of in the init of human.

https://github.com/pg2455/covid_p2p_simulation/pull/81
https://www.zmescience.com/science/viral-load-coronavirus-03042020/
https://www.newscientist.com/article/2238819-does-a-high-viral-load-or-infectious-dose-make-covid-19-worse/
https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30237-1/fulltext
https://www.thelancet.com/journals/laninf/article/PIIS1473-3099(20)30232-2/fulltext

File: simulator.py (hunk 3 above)
Comment (teganmaharaj) [reply simulator.py:718]: I don't think it will miss that; it will be taken care of by the loops of other humans. Call the current self Zach, who is infectious. He goes through a loop with Yiu and Xena. Then we do loops where Yiu and Xena each become "self", and they will see Zach as "h", who will infect them.

As long as we don't remove humans from locations when they're done looping.

File: simulator.py (hunk 1 above)
Comment (pg2455) [reply simulator.py:78]: Yeah! I think it's good to have them. really impressive you have nailed down all these parameters. 💯

File: simulator.py (hunk 3 above)
Comment (pg2455) [reply simulator.py:718]: `[A] --> [A, B*] --> [A, B*, C]` by the above logic, B will infect C, but not A.

Comment (pg2455) [reply simulator.py:718]: [We had a long discussion on slack on this issue](https://covid19-task-force.slack.com/archives/C011C7KGPJB/p1586460189059800?thread_ts=1586458139.045000&cid=C011C7KGPJB).
